Tests/update.Tests.ps1

```diff
@@ -230,30 +264,171 @@ Describe 'prism update' {
+            "Version": "1.*"
+        },
+        {
+            "Name": "Whiskey",
+            "Version": "0.*"
+        }
+    ]
+}
+'@
+        GivenLockFile  @"
+{
+    "PSModules":  [
+        {
+            "name":  "NoOp",
+            "version":  "1.0.0",
+            "repositorySourceLocation":  "$($script:defaultLocation)"
+        },
+        {
+            "name":  "Carbon",
+            "version":  "1.0.0",
+            "repositorySourceLocation":  "$($script:defaultLocation)"
+        },
+        {
+            "name":  "Whiskey",
+            "version":  "0.14.0",
+            "repositorySourceLocation":  "$($script:defaultLocation)"
+        }
+    ]
+}
+"@
+        WhenLocking -Name ('NoOp', 'Carbon', 'Bogus')
+        # Latest NoOp version 1 should be >= 1.1.0, set at 1.0.0 for testing. This should get updated.
+        # Latest Carbon version 1 should be 1.9.0, set at 1.0.0 for testing. This should get updated.
+        # Latest Whiskey version 0 should be >= 0.61.0, set at 0.14.0 for testing. This should remain unchanged.
+        $expectedCarbonModule =
+            Find-Module -Name 'Carbon' -AllVersions | Where-Object 'Version' -like '1.*' | Select-Object -First 1
+        $carbonV0DefaultLocation =
+            Get-PSRepository -Name $expectedCarbonModule.Repository | Select-Object -ExpandProperty 'SourceLocation'
+        ThenLockFileIs ([pscustomobject]@{
+            PSModules = @(
+                [pscustomobject]@{
+                    name = 'NoOp';
+                    version = $script:latestNoOpModule.Version;
+                    repositorySourceLocation = $script:defaultLocation;
+                },
+                [pscustomobject]@{
+                    name = 'Carbon';
+                    version = $expectedCarbonModule.Version;
+                    repositorySourceLocation = $carbonV0DefaultLocation;
+                },
+                [pscustomobject]@{
+                    name = 'Whiskey';
+                    version = '0.14.0';
+                    repositorySourceLocation = $script:defaultLocation;
+                }
+            )
+        })
+    }
+
+    It 'should ignore modules that are not currently in the lock file when updating a specific set of modules' {
+        GivenPrismFile @'
+{
+    "PSModules": [
+        {
+            "Name": "NoOp",
+            "Version": "1.*"
+        },
+        {
+            "Name": "Carbon",
+            "Version": "1.*"
+        },
+        {
+            "Name": "Whiskey",
+            "Version": "0.*"
+        }
+    ]
+}
+'@
+        GivenLockFile  @"
+{
+    "PSModules":  [
+        {
+            "name":  "NoOp",
+            "version":  "1.0.0",
+            "repositorySourceLocation":  "$($script:defaultLocation)"
+        },
+        {
+            "name":  "Carbon",
+            "version":  "1.0.0",
+            "repositorySourceLocation":  "$($script:defaultLocation)"
+        }
+    ]
+}
+"@
+        WhenLocking -Name ('NoOp', 'Carbon')

```

Comment:
Might be better to rename this `WhenLocking` function to `prism` so that we can write the tests as:

```
prism update NoOp,  Carbon
```

And just pass all the parameters directly through to Prism.